torchvision/transforms/transforms.py

```diff
@@ -377,6 +378,13 @@ def __init__(self, size, padding=0, pad_if_needed=False):
             self.size = (int(size), int(size))
         else:
             self.size = size
+
+        if isinstance(padding, numbers.Number):
```

Comment:
@KovenYu good catch on padding type! However I have an impression that a similar stuff is done in [funcitional](https://github.com/pytorch/vision/blob/master/torchvision/transforms/functional.py#L261) ? 
Would not it be more simple to use `pad_if_needed` in condition`if any(np.array(self.padding) > 0):`, what do you think ?

Reply:
@vfdev-5 yes you're right! it should be simplified. 

I do so simply because in **functional**, if you give a 2-element tuple (a, b) as **padding**, it will interpret it as (width, height). 

this is not natural because if you specify RandomCrop(size=(128, 32), padding=(16, 4)), you are probably expecting the height 128 padded with 16 and width 32 padded with 4. but in fact, it is inverse. So this patch of codes are intended for exchanging the positions of these two elements, although it looks, yeah, a little messy ..

what do you think of this problem? if you feel this problem worth fixing, I think we can add some code in __call__ for the exchanging task. anyway, I agree we should remove this patch in __init__.

Reply:
@KovenYu I see better the problem, thanks! 
The main bug is in the condition `if self.padding > 0:` and it should be fixed.

Next, you propose an update of padding argument acceptable value as actually, according to the doc of [`RandomCrop`](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py#L367) it says:
```
padding (int or sequence, optional): Optional padding on each border
            of the image. Default is 0, i.e no padding. If a sequence of length
            4 is provided, it is used to pad left, top, right, bottom borders
            respectively.
```
so, we logically should not provide a tuple like `(16, 4)`. But as you are right in functional there is an option to set padding as [2-element tuple](https://github.com/pytorch/vision/blob/master/torchvision/transforms/functional.py#L220).
But as it is said in the doc on padding:
> If tuple of length 2 is provided this is the padding on left/right and top/bottom respectively

So it is correct to get in your example `RandomCrop(size=(128, 32), padding=(16, 4))` the result you describe.

Maybe we need to uniformize all these arguments and fix the condition by setting `padding=None`  instead of `padding=0` as no padding...
What do you think ?

Reply:
@KovenYu At first, I think we can improve the doc of `RandomCrop` adding `If tuple of length 2 is provided this is the padding on left/right and top/bottom respectively.` taken from **functional** and concerning the condition fix I think we can wait maintainer's decision, cc @fmassa @soumith.
Your solution works but personnally I would just change `padding=0` by `padding=None` and check if it is not None. Let's see what they say...

However, I think there is also something strange in the current API with `padding=0` and boolean `pad_if_needed`. I see the purpose of the second argument, but the chosen name and the idea to crop to a larger image seems little unnatural imho...

Reply:
I'm fine with making `padding=None` by default, and checking that `if padding is not None`.
About changing the order of `height`/`width`, let's not change it now but just improve the documentation.